presto-main/src/main/java/io/prestosql/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -428,11 +432,25 @@ static Integer columnSize(Type type)
         if (type.equals(DATE)) {
             return 14; // +5881580-07-11 (2**31-1 days)
         }
-        if (type.equals(TIMESTAMP)) {
-            return 15 + 8;
-        }
-        if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
-            return 15 + 8 + 6;
+        if (type instanceof TimestampType) {
+            // 1 digit for year sign
+            // 5 digits for year
+            // 15 digits for "-MM-DD HH:MM:SS"
+            // min(p, 1) for the fractional second period (i.e., no period if p == 0)
+            // p for the fractional digits
+            int precision = ((TimestampType) type).getPrecision();
+            return 1 + 5 + 15 + min(precision, 1) + precision;
+        }
+        if (type instanceof TimestampWithTimeZoneType) {
+            // 1 digit for year sign
+            // 5 digits for year
```

Comment:
I believe it's 5, but please check

Reply:
timestamp can definitely acommodate 6 digit years (and we have tests for that), 
but i don't remember whether i checked the max timestamp and whether it could have 7.
IIRC  _timestamp with time zone_ ends up at 5 digit years